src/metrics.jl

```diff
@@ -214,6 +249,29 @@ eval_end(::Euclidean, s) = sqrt(s)
 euclidean(a::AbstractArray, b::AbstractArray) = evaluate(Euclidean(), a, b)
 euclidean(a::Number, b::Number) = evaluate(Euclidean(), a, b)
 
+# PeriodicEuclidean
+Base.eltype(d::PeriodicEuclidean) = eltype(d.periods)
+@inline parameters(d::PeriodicEuclidean) = d.periods
+@inline function eval_start(d::PeriodicEuclidean, a::AbstractArray, b::AbstractArray)
+    zero(result_type(d, a, b))
+end
+@inline function eval_op(d::PeriodicEuclidean, ai, bi, pi)
+    s = abs(ai - bi)
+    s = mod(s, pi)
```

Comment:
Are we sure `s` won't change type here? In doubt, maybe better use different names for the variable (same below).